### ISSUE TITLE
Problems encountered during data preprocessing！

### ISSUE THREAD

**Issue by telee6789** [zhangshuming0668/BrepMFR#1]
I am currently encountering an issue with the preprocessing of the bin file, as the process described

 cannot be reproduced. In your article, the dimension of the UV-Grid Point of the curve is mentioned as 6,
 but the dimension of the data read from the bin file is 7.

Could you kindly provide the Python script for the bin file preprocessing process?

 It would be greatly appreciated!

mentioned as 6
![image](https://github.com/zhangshuming0668/BrepMFR/assets/8375346/b30429ff-c134-4cdd-a2ca-c181008e66cb)

mentioned as 7
![image](https://github.com/zhangshuming0668/BrepMFR/assets/8375346/46143202-056a-4917-a7e9-4a94590937ba)

**Comment by youwei1** [comment]
This link may be helpful to you [https://github.com/AutodeskAILab/UV-Net/blob/main/process/solid_to_graph.py](url)

**Comment by Redsparkfish** [comment]
It seems to me that there's problem with the dataset CADSynth. For example, 00098525.stp is a simple cylinder with a hole. It can be counted that there are 6 edges of 2 types (line and circle). But the graph loaded from 00098525.bin has 8 edges, and they are all of the same type. Some of them are obviously duplicated. I believe this is not the only case.

**Comment by msattari90** [comment]
I am also facing the same issue! Without the step->bin script, it is not possible to use the model!

**Comment by aryanseth-datum** [comment]
Hey all, I found this adjacent file from Autodesk's UV-Net library that mentions converting step to the BIN and because the BrepfMFR also ran UV-NET on it along with other datasets, im inclined to believe this is the same technique the authors used

https://github.com/AutodeskAILab/UV-Net/blob/main/process/README.md

here is the original github link: https://github.com/AutodeskAILab/UV-Net/tree/main that talks about he DGL format

**Comment by Fdioa** [comment]
@aryanseth-datum I tried this stp -> bin method, but the model is not difficult to load the bin file converted by UV-net.

**Comment by aryanseth-datum** [comment]
@Fdioa  What do you mean it is not difficult to load the bin file? Are you saying the program above didn't work?

**Comment by Fdioa** [comment]
@aryanseth-datum  yes, it didn't work for BrepMFR. I tried to reproduce the conversion file that conforms to the BrepMFR format. Although I could input the BrepMFR model correctly, the accuracy was very low due to significant differences in details. 

[stp_to_bin.py](https://github.com/user-attachments/files/23530914/stp_to_bin.py)
